connectors/sources/github.py

```diff
@@ -215,20 +369,297 @@ async def validate_config(self):
+
+                    # Converting the local repository names to username/repo_name format.
+                    if "/" not in repo_name:
+                        repo_name = f"{self.user.get('login')}/{repo_name}"
+
+                    repo_object = self.foreign_repos.get(
+                        repo_name
+                    ) or self.user_repos.get(repo_name)
+                    if not repo_object:
+                        repo_object = await self.github_client.get_response(
+                            resource=ENDPOINTS["REPO"].format(repo_name=repo_name)
+                        )
+
+                    document = self.prepare_document(
+                        github_document=repo_object,
+                        schema=REPO_SCHEMA,
+                    )
+                    document["owner"] = repo_object["owner"]["login"]  # pyright: ignore
+                    document["type"] = ObjectType.REPOSITORY.value
+                    yield document
+        except UnauthorizedException:
+            raise
+        except Exception as exception:
+            logger.warning(
+                f"Something went wrong while fetching the repository. Exception: {exception}"
+            )
+
+    async def fetch_pulls(self, repo_name):
+        try:
+            async for pull in self.github_client.get_paginated_response(
+                resource=ENDPOINTS["PULLS"].format(
+                    repo_name=repo_name, page_size=PAGE_SIZE
+                )
+            ):
+                document = self.prepare_document(
+                    github_document=pull, schema=PULL_REQUEST_SCHEMA
+                )
+                comments, review_comments, reviews = [], [], []
+                async for review_comment in self.github_client.get_paginated_response(
+                    ENDPOINTS["REVIEW_COMMENTS"].format(
+                        repo_name=repo_name, number=pull["number"], page_size=PAGE_SIZE
+                    )
+                ):
+                    review_comments.append(
+                        {
+                            "author": review_comment["user"]["login"],
+                            "body": review_comment["body"],
+                        }
+                    )
+                async for comment in self.github_client.get_paginated_response(
+                    ENDPOINTS["COMMENTS"].format(
+                        repo_name=repo_name, number=pull["number"], page_size=PAGE_SIZE
+                    )
+                ):
+                    comments.append(
+                        {"author": comment["user"]["login"], "body": comment["body"]}
+                    )
+                async for review in self.github_client.get_paginated_response(
+                    ENDPOINTS["REVIEWS"].format(
+                        repo_name=repo_name, number=pull["number"], page_size=PAGE_SIZE
+                    )
+                ):
+                    reviews.append(
+                        {
```

Comment:
nit: Could also live inside a method `_review_doc(review)` to improve readability